File: backend/Modules/Professionals/Modules.Professionals.Features/GetProfessional/GetProfessionalDto.cs
using Modules.Common.Features.ValueObjects;
using Modules.Professionals.Domain.Enums;
using Modules.Professionals.Features.GetServices;
using Modules.Professionals.Features.GetSpecializations;

namespace Modules.Professionals.Features.GetProfessional;

public sealed record GetProfessionalDto(
    Guid Id,
    Guid UserId,
    string FirstName,
    string LastName,
    string Email,
    string PhoneNumber,
    string DateOfBirth,
    string Gender,
    Address Address,
    SpecializationDto Specialization,
    List<ServiceDto> Services,
    int Experience,
    int VisitPrice,
    string? Bio,
    string? ProfilePictureUrl,
    VerificationStatus VerificationStatus,
    double? DistanceKm = null);

File: frontend/src/features/professional/hooks/GetProfessional/GetProfessionalDto.ts
import type { Address } from "@/features/auth";
import type { VerificationStatus } from "@/features/admin/types/adminTypes";

export interface GetProfessionalDto {
  id: string;
  userId: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  dateOfBirth: string;
  gender: string;
  address: Address;
  specialization: string;
  services: string[];
  experience: number;
  visitPrice: number;
  bio: string;
  isVerified: boolean;
  profilePictureUrl: string;
  verificationStatus: VerificationStatus;
  distanceKm?: number | null;
}
